src/wallet/api.rs

```diff
@@ -660,7 +660,11 @@ impl Wallet {
         let all_utxos = if let Some(utxos) = utxos {
             utxos.clone()
         } else {
-            self.get_all_utxo()?
+            log::info!("FETCHING UTXO From Cache!");
+            self.store.utxo_cache
+            .values()
+            .map(|(utxo, _spend_info)| utxo.clone())
+            .collect()
         };
 
         let processed_utxos = all_utxos
```

Comment:
This logic is now redundant. all the utxos are already processed in the cache. No need to process again.